installer.sh.in

```diff
@@ -846,9 +867,32 @@ configure_wifi() {
         echo "  auth_alg=SHARED" >> ${_wpasupconf%.conf}-${dev}.conf
         echo "}" >> ${_wpasupconf%.conf}-${dev}.conf
     else
-        wpa_passphrase "$ssid" "$pass" >> ${_wpasupconf%.conf}-${dev}.conf
+        echo "network={" >> ${_wpasupconf%.conf}-${dev}.conf
+        echo "  ssid=\"$ssid\"" >> ${_wpasupconf%.conf}-${dev}.conf
+        echo "  scan_ssid=1" >> ${_wpasupconf%.conf}-${dev}.conf
+        echo "  key_mgmt=WPA-PSK" >> ${_wpasupconf%.conf}-${dev}.conf
+        echo "  psk=\"$pass\"" >> ${_wpasupconf%.conf}-${dev}.conf
+        echo "}" >> ${_wpasupconf%.conf}-${dev}.conf
```

Comment:
This.... this already is what is done. Why are you doing this?

Reply:
The reason why I replaced "wpa_passphrase ...." by hardcoding it is because it's consistent with how WEP is configured.  I mean it's not a big deal, you can change that.

Reply:
I don't understand what you are saying.

Reply:
Ok, so the way the WEP configuration is modified is like this

    echo "network={" >> ${_wpasupconf%.conf}-${dev}.conf
            echo "  ssid=\"$ssid\"" >> ${_wpasupconf%.conf}-${dev}.conf
            echo "  wep_key0=\"$pass\"" >> ${_wpasupconf%.conf}-${dev}.conf
            echo "  wep_tx_keyidx=0" >> ${_wpasupconf%.conf}-${dev}.conf
            echo "  auth_alg=SHARED" >> ${_wpasupconf%.conf}-${dev}.conf
    echo "}" >> ${_wpasupconf%.conf}-${dev}.conf

The configuration for WPA used to be this

    wpa_passphrase "$ssid" "$pass" >> ${_wpasupconf%.conf}-${dev}.conf

So I changed that to this

    echo "  scan_ssid=1" >> ${_wpasupconf%.conf}-${dev}.conf    
    echo "  key_mgmt=WPA-PSK" >> ${_wpasupconf%.conf}-${dev}.conf
    echo " psk=\"$pass\"" >> ${_wpasupconf%.conf}-${dev}.conf
    echo "}" >> ${_wpasupconf%.conf}-${dev}.conf

I mean this change is not a big deal, I don't care if it's merged or not. But the reason why I changed it is that it makes the WPA configuration consistent with the WEP configuration. Basically, it just looks nicer IMO. It doesn't do anything differently, I just think it makes the code somewhat more readable.